src/pysdmx/io/xml/sdmx_two_one/writer/metadata_writer.py

```diff
@@ -0,0 +1,276 @@
+
+    return outfile
+
+
+def __write_item(item: Item, indent: str) -> str:
+    head = f"{ABBR_STR}:" + type(item).__name__
+
+    data = __write_nameable(item, add_indent(indent))
+    outfile = f'{indent}<{head}{data["Attributes"]}>'
+    outfile += export_intern_data(data, add_indent(indent))
+    outfile += f"{indent}</{head}>"
+    # if self.parent is not None:
+    #     indent_par = add_indent(indent)
+    #     indent_ref = add_indent(indent_par)
+    #     outfile += f"{indent_par}<{ABBR_STR}:Parent>"
+    #     if isinstance(self.parent, Item):
+    #         text = self.parent.id
+    #     else:
+    #         text = self.parent
+    #     outfile += f'{indent_ref}<Ref id="{text}"/>'
+    #     outfile += f"{indent_par}</{ABBR_STR}:Parent>"
+    return outfile
+
+
+def __write_item_scheme(item_scheme: ItemScheme, indent: str) -> str:
+
+    label = f"{ABBR_STR}:{type(item_scheme).__name__}"
+
+    data = __write_maintainable(item_scheme, indent)
+
+    data["Attributes"] += f" isPartial={str(item_scheme.is_partial).lower()!r}"
+
+    outfile = ""
+
+    attributes = data.get("Attributes") or ""
+    attributes = attributes.replace("'", '"')
+
+    outfile += f"{indent}<{label}{attributes}>"
+
+    outfile += export_intern_data(data, indent)
+
+    for item in item_scheme.items:
+        outfile += __write_item(item, add_indent(indent))
+
+    outfile += f"{indent}</{label}>"
+
+    return outfile
+
+
+def __write_metadata_element(
+    package: Dict[str, Any], key: str, prettyprint: object
+) -> str:
+    """Writes the metadata element to the XML file.
+
+    Args:
+        package: The package to be written
+        key: The key to be used
+        prettyprint: Prettyprint or not
+
+    Returns:
+        A string with the metadata element
+    """
+    outfile = ""
+    nl = "\n" if prettyprint else ""
+    child2 = "\t\t" if prettyprint else ""
+
+    base_indent = f"{nl}{child2}"
+
+    if key in package:
+        outfile += f"{base_indent}<{ABBR_STR}:{MSG_CONTENT_PKG[key]}>"
+        for item_scheme in package[key].values():
+            outfile += __write_item_scheme(
+                item_scheme, add_indent(base_indent)
+            )
+        outfile += f"{base_indent}</{ABBR_STR}:{MSG_CONTENT_PKG[key]}>"
+
+    return outfile
+
+
+def generate_structures(content: Dict[str, Any], prettyprint: bool) -> str:
+    """Writes the structures to the XML file.
+
+    Args:
+        content: The Message Content to be written
+        prettyprint: Prettyprint or not
+
+    Returns:
+        A string with the structures
+    """
+    nl = "\n" if prettyprint else ""
+    child1 = "\t" if prettyprint else ""
+
+    outfile = f"{nl}{child1}<{ABBR_MSG}:Structures>"
+
+    for key in MSG_CONTENT_PKG:
+        outfile += __write_metadata_element(content, key, prettyprint)
+
+    outfile += f"{nl}{child1}</{ABBR_MSG}:Structures>"
+
+    return outfile
+
+
+def export_intern_data(data: Dict[str, Any], indent: str) -> str:
```

Comment:
Apparently, only `generate_structure` is imported? If so, should this method be public?

Reply:
Yes, changing the rest of methods to private (with __ at the beginning)